### DISCUSSION TITLE
Add a flag to warn users about a duplicate value for a field

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5819] [Feature Requests]
This is something that could be applied within the schema configuration, then would be used everywhere else.

For instance, you would mark `altCatalogNumber` as a field that should warn if it is not unique.

When creating new records, if the field's value is used in an existing record it would "flag" before saving or in the interface. This would be displayed as a flag in the WorkBench when validating as well ideally.

> During the migration mapping for our herbarium we were asked if we could flag a particular field (field book number) as being a duplicate, but not block the user from uploading the data. In this example, there are some cases where a field book number could be re-used for a new collection object, but the user should double-check before upload because it complicates the data.
> 
> It would be great if we could add a flag in the schema config for any field so that when a user saves a record via a data entry form they get a popup window that lists any fields with this flag that are duplicates, along with some text like "Are you sure you wish to continue?". This would also need to be compatible with workbench uploads, perhaps either as a new colour-coding for cells when a user does validation, or a popup warning before uploading with a list of the field name and field value for any that are duplicates.

**Requested By:** Dan Baker at CSIRO ([Asana](https://app.asana.com/0/1201972579963460/1204663198443048/f))

**Comment by grantfitzsimmons** [comment]
WorkBench uses back-end error reporting, so this is something that would be globally applied.

This would need us to engineer a new form system for validating independent of the browser's per Jason

**Comment by melton-jason** [comment]
For further clarification, we would not necessarily need to engineer a new form system. 

**Frontend**
If we want the warning to be displayed like the save blockers are , we would utilize the browser's built-in [setting custom form validity](https://developer.mozilla.org/en-US/docs/Web/API/HTMLObjectElement/setCustomValidity). 

The 'new system' would probably just be extending save blockers to support these warnings and modify our saving behavior**. 

**Backend**
The work on the 'backend' would be for Workbench uploads.The Backend _does_ have a system for checking Uniqueness Rules, but another exception would need to be defined.

**Comment by grantfitzsimmons** [comment]
Geneva would like this as well.

**Comment by maxpatiiuk** [comment]
@grantfitzsimmons is this a duplicate of #2712?

**Comment by grantfitzsimmons** [comment]
@maxpatiiuk  No– this could be aided by the work in #2712 though.

**Comment by bronwyncombs** [comment]
College of Idaho would like this too
